enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/RegisteredPipeBuilder.scala

```diff
@@ -114,6 +114,12 @@ class RegisteredPipeBuilder(fallback: PipeBuilder,
       case _: Selection =>  // selection relies on inner expressions to interact with variables
         fallback.build(plan, source)
 
+      case _: OptionalExpand =>
```

Comment:
OptionalExpandPipe writes to execution contexts and should not be handled by the fallback.